### DISCUSSION TITLE
posts/JS-Reference-Error/

### DISCUSSION THREAD

**Discussion by giscus** [da-in/da-in.github.io#6] [General]
# posts/JS-Reference-Error/

이걸 이렇게 고민하게 될 줄 몰랐건만… 결과적으로 1500명 풀방의 오픈카톡방을 활활 타오르게 만들었던 재미있는 주제…🔥

https://da-in.github.io/posts/JS-Reference-Error/

**Comment by devhoodit** [comment]
첫번째 에러문은  let이 아직 선언이 되지 않았고, 괄호로 감싸 지역으로 들어가기 전에 let을 통해 선언 되었고 (초기화 되지 않은 상태)
지역으로 들어가면 inner scope에서 outer scope로 찾는 과정에서 local에 변수가 존재하지 않으니 global scope로 이동해서 변수를 찾는 과정으로 가고, global에서 let을 통해 선언이 된 (초기화가 되지 않은) 변수를 만나 그런것 같습니다
